src/monodroid/jni/basic-utilities.cc

```diff
@@ -284,69 +284,73 @@ BasicUtilities::monodroid_strfreev (char **str_array)
 char **
```

Comment:
If we're gonna "improve" things, should we consider changing the return type of `monodroid_strsplit()` from `char**` to `vector<string[max_tokens]>`?  (Is that even an express-able construct?)

Reply:
We can't do that since that would be an ABI break. `monodroid_strsplit` is a p/invoke, and also we pass its return value to Mono when [initializing the JIT](https://github.com/xamarin/xamarin-android/blob/617500c92d989c87641db7010595f296eeffcfde/src/monodroid/jni/monodroid-glue.cc#L823).